packages/@sfcc-core/core/src/index.ts

```diff
@@ -1,18 +1,33 @@
-/*
-    Copyright (c) 2020, salesforce.com, inc.
-    All rights reserved.
-    SPDX-License-Identifier: BSD-3-Clause
-    For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
-*/
-/*
-   SFRA Core Code
-    */
+import {
+    ApiConfig,
+    Logger,
+    Extension,
+    Service,
+    ExtensionFactory,
+    LogLevel,
+} from './types';
 
 export const LOGGER_KEY = Symbol('Logger Service');
 export const API_EXTENSIONS_KEY = Symbol('API Extensions');
 
+class UnknownServiceError extends Error {
+    __proto__: Error;
+
+    constructor(m: string) {
+        const trueProto = new.target.prototype;
+        super(m);
+        this.__proto__ = trueProto;
+    }
+}
+
 class Core {
-    //singletons (logger, etc)
+    _services: { [key: string]: Service };
+    _extensions: { [key: string]: Array<ExtensionFactory> };
+    _factoryExtensions: { [key: string]: Array<ExtensionFactory> };
+    _factoryServices: { [key: string]: Service };
+
```

Comment:
We can use `private` keyword before member declarations.

Reply:
Good call